confignode/src/main/java/org/apache/iotdb/confignode/manager/load/balancer/RouteBalancer.java

```diff
@@ -41,22 +45,64 @@ public class RouteBalancer {
 
   private final IManager configManager;
 
+  private final LazyRandomRouter lazyRandomRouter;
+
   public RouteBalancer(IManager configManager) {
     this.configManager = configManager;
+    this.lazyRandomRouter = new LazyRandomRouter();
   }
 
   public Map<TConsensusGroupId, TRegionReplicaSet> genLatestRegionRouteMap(
       List<TRegionReplicaSet> regionReplicaSets) {
-    return genRouter().genLatestRegionRouteMap(regionReplicaSets);
+    List<TRegionReplicaSet> schemaRegionGroups = new ArrayList<>();
+    List<TRegionReplicaSet> dataRegionGroups = new ArrayList<>();
+
+    regionReplicaSets.forEach(
+        regionReplicaSet -> {
+          switch (regionReplicaSet.getRegionId().getType()) {
+            case SchemaRegion:
+              schemaRegionGroups.add(regionReplicaSet);
+              break;
+            case DataRegion:
+              dataRegionGroups.add(regionReplicaSet);
+              break;
+          }
+        });
+
+    // Generate SchemaRegionRouteMap
+    Map<TConsensusGroupId, TRegionReplicaSet> result =
+        genRouter(TConsensusGroupType.SchemaRegion).genLatestRegionRouteMap(schemaRegionGroups);
+    // Generate DataRegionRouteMap
+    result.putAll(
+        genRouter(TConsensusGroupType.DataRegion).genLatestRegionRouteMap(dataRegionGroups));
+    return result;
   }
 
-  private IRouter genRouter() {
+  private IRouter genRouter(TConsensusGroupType groupType) {
     String policy = ConfigNodeDescriptor.getInstance().getConf().getRoutingPolicy();
-    if (policy.equals(leaderPolicy)) {
-      return new LeaderRouter(
-          getLoadManager().getAllLeadership(), getLoadManager().getAllLoadScores());
-    } else {
-      return new LoadScoreGreedyRouter(getLoadManager().getAllLoadScores());
+    switch (groupType) {
+      case SchemaRegion:
+        if (policy.equals(leaderPolicy)) {
+          return new LeaderRouter(
+              getLoadManager().getAllLeadership(), getLoadManager().getAllLoadScores());
+        } else {
+          return new LoadScoreGreedyRouter(getLoadManager().getAllLoadScores());
+        }
+      case DataRegion:
+      default:
+        if (ConfigNodeDescriptor.getInstance()
+            .getConf()
+            .getDataRegionConsensusProtocolClass()
+            .equals(ConsensusFactory.MultiLeaderConsensus)) {
+          // Latent router for MultiLeader consensus protocol
+          lazyRandomRouter.updateUnknownDataNodes(getLoadManager().getUnknownDataNodes(-1));
```

Comment:
Why not initialize a globally unique variable like `lazyRandomRouter`?
I think this method may be called frequently, so many objects will be created and GC by the jvm

Reply:
Tanks for your advice, I'll optimize these code in next PR~